packages/react-native/Libraries/Text/TextInput/RCTBaseTextInputView.mm

```diff
@@ -314,6 +314,16 @@ - (void)setTextContentType:(NSString *)type
       }];
     }
 #endif
+      
+#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_15_0
+    if (@available(iOS 15.0, *)) {
+      [mutableContentTypeMap addEntriesFromDictionary:@{
+        @"dateTime" : UITextContentTypeDateTime,
+        @"flightNumber" : UITextContentTypeFlightNumber,
+        @"shipmentTrackingNumber" : UITextContentTypeShipmentTrackingNumber,
+      }];
+    }
+#endif
```

Comment:
Can we order these from old to new (so first IOS15+, then iOS17+). 

Also, we're always targeting the latest iOS SDK, so we don't need these ifdefs anymore I beleive.

Reply:
@javache 

I changed order and removed ifdefs only for iOS15 as removing those for iOS17 was causing pipeline to fail